WooCommerce/Classes/ViewRelated/Products/Edit Product/Subscription/Trial/SubscriptionTrialViewController.swift

```diff
@@ -0,0 +1,180 @@
+import UIKit
+import SwiftUI
+import Kingfisher
+import enum Yosemite.SubscriptionPeriod
+
+// MARK: Hosting Controller
+
+/// Hosting controller that wraps a `SubscriptionTrialView` view.
+///
+final class SubscriptionTrialViewController: UIHostingController<SubscriptionTrialView> {
+    init(viewModel: SubscriptionTrialViewModel) {
+        super.init(rootView: SubscriptionTrialView(viewModel: viewModel))
+        title = viewModel.title
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: Views
+
+/// Renders a list of components in a composite product
+///
+struct SubscriptionTrialView: View {
+
+    /// View model that directs the view content.
+    ///
+    @StateObject var viewModel: SubscriptionTrialViewModel
+
+    /// Scale of the view based on accessibility changes
+    ///
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    var body: some View {
+        ScrollView {
+            VStack(alignment: .leading, spacing: 0) {
+                // Duration
+                Group {
+                    TitleAndTextFieldRow(title: Localization.duration,
+                                         placeholder: "0",
+                                         text: $viewModel.trialLength,
+                                         keyboardType: .asciiCapableNumberPad,
+                                         inputFormatter: IntegerInputFormatter())
+
+                    Divider()
+                        .padding(.leading, Layout.margin)
+                }
+
+                period
+
+                // Validation error
+                Group {
+                    ValidationErrorRow(errorMessage: viewModel.errorMessage)
+
+                    Divider()
+                        .padding(.leading, Layout.margin)
+                }
+                .renderedIf(!viewModel.isInputValid)
+
+                // Subtitle
+                Text(Localization.freeTrialSubtitle)
+                    .foregroundColor(Color(.textSubtle))
+                    .subheadlineStyle()
+                    .padding(Layout.margin)
```

Comment:
I agree with Hafiz's [comment](https://github.com/woocommerce/woocommerce-ios/pull/11173#issuecomment-1813958902) - how about we display the subtitle only when there is no validation error?

```suggestion
                // Subtitle
                Text(Localization.freeTrialSubtitle)
                    .foregroundColor(Color(.textSubtle))
                    .subheadlineStyle()
                    .padding(Layout.margin)
                    .renderedIf(viewModel.isInputValid)
```

Reply:
Done in d4250cfa7dfade9da19d7fe7ea5f2714d071a424